std/typecons.d

```diff
@@ -6896,3 +6896,222 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+/++
+ + Type constructor for head-const variables.
+ +
+ + Head-const variables cannot be directly mutated or rebound, but references
+ + reached through the variable are typed with their original mutability.
+ + It is equivalent to $(D final) variables in D1 and Java, as well as
```

Comment:
@andralex 
> I think therefore `Final` would be a better choice than `HeadConst`

Someone needs to make an executive decision about the name. It has already been debated (and changed once), with about a 50% / 50% split of votes:

[For `HeadConst`](https://github.com/dlang/phobos/pull/2881#issuecomment-70852797): **ZombineDev**, **JackStouffer**, **schuetzm**, **Dicebot**

[For `Final`](https://github.com/dlang/phobos/pull/3862#issuecomment-194334468): **JakobOvrum**, **DmitryOlshansky**, **tsbockman** (me), **andralex**

(Click the links if you want to skim the earlier discussion.)